tests/cql_query_test.cc

```diff
@@ -4261,3 +4261,106 @@ SEASTAR_TEST_CASE(test_rf_expand) {
+                "    AND crc_check_chance = 1\n"
+                "    AND dclocal_read_repair_chance = 0.1\n"
+                "    AND default_time_to_live = 0\n"
+                "    AND gc_grace_seconds = 864000\n"
+                "    AND max_index_interval = 2048\n"
+                "    AND memtable_flush_period_in_ms = 0\n"
+                "    AND min_index_interval = 128\n"
+                "    AND read_repair_chance = 0\n"
+                "    AND speculative_retry = '99.0PERCENTILE';\n";
+        e.execute_cql(create_table_with_compund_key).get();
+        schema = e.local_db().find_schema("ks", "cf1");
+
+        std::ostringstream ss1;
+        schema->describe(ss1);
+        BOOST_CHECK_EQUAL(ss1.str(), create_table_with_compund_key);
+    });
+}
+
+SEASTAR_TEST_CASE(test_describe_view_schema) {
+    return do_with_cql_env_thread([] (cql_test_env& e) {
+        e.execute_cql("CREATE TABLE ks.cf (\n"
+                "    pk blob,\n"
+                "    col1 blob,\n"
+                "    col2 blob,\n"
+                "    PRIMARY KEY (pk, col1))").get();
+        std::string create_materialized_view = "CREATE MATERIALIZED VIEW ks.cf_view AS\n"
+                "    SELECT col1, pk\n"
+                "    FROM cf\n"
+                "    WHERE pk IS NOT null AND col1 IS NOT null\n"
+                "    PRIMARY KEY (col1, pk)\n"
+                "    WITH CLUSTERING ORDER BY (pk ASC)\n"
+                "    AND bloom_filter_fp_chance = 0.01\n"
+                "    AND caching = {'keys': 'ALL','rows_per_partition': 'ALL'}\n"
+                "    AND comment = ''\n"
+                "    AND compaction = {'class': 'SizeTieredCompactionStrategy'}\n"
+                "    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}\n"
+                "    AND crc_check_chance = 1\n"
+                "    AND dclocal_read_repair_chance = 0.1\n"
+                "    AND default_time_to_live = 0\n"
+                "    AND gc_grace_seconds = 864000\n"
+                "    AND max_index_interval = 2048\n"
+                "    AND memtable_flush_period_in_ms = 0\n"
+                "    AND min_index_interval = 128\n"
+                "    AND read_repair_chance = 0\n"
+                "    AND speculative_retry = '99.0PERCENTILE';\n";
+        e.execute_cql(create_materialized_view).get();
+        auto schema = e.local_db().find_schema("ks", "cf_view");
+        std::ostringstream ss;
+
+        schema->describe(ss);
+        BOOST_CHECK_EQUAL(ss.str(), create_materialized_view);
+        std::string create_index = "CREATE INDEX cf_index ON ks.cf(col2);\n";
+        e.execute_cql(create_index).get();
+        schema = e.local_db().find_schema("ks", "cf_index_index");
+
+        std::ostringstream ss1;
+        schema->describe(ss1);
+        BOOST_CHECK_EQUAL(ss1.str(), create_index);
+    });
```

Comment:
Please test at the end of this test that describe() of "cf", (not "cf_view" or "cf_index_index"). I don't think it's obvious whether describe of a table *with* materialized views should list them or not - but since in the code we decided NOT to list them, the test should verify this: describe() of cf, despite it having a materialized view and an index, should contain CQL to just create cf, not its view.

Reply:
good point, I'll add a check for that as well